### PR TITLE
macOS port

### PR DESCRIPTION
**Pull request by matijaskala**
These changes are apparently required on macOS

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -142,7 +142,10 @@ else()
     target_compile_options(nvi PRIVATE -Wno-pointer-sign)
 endif()
 
-target_link_libraries(nvi PRIVATE ${CURSES_LIBRARY} ${TERMINFO_LIBRARY})
+target_link_libraries(nvi PRIVATE ${CURSES_LIBRARY})
+if(TERMINFO_LIBRARY)
```
Comment (lichray) [inline CMakeLists.txt:146]: Is this necessary?

File: common/exf.c [hunk 2]
```diff
@@ -199,7 +199,14 @@ file_init(SCR *sp, FREF *frp, char *rcv_name, int flags)
 		if (!LF_ISSET(FS_OPENERR))
 			F_SET(frp, FR_NEWFILE);
 
+#if defined HAVE_STRUCT_STAT_ST_MTIMESPEC
+		ep->mtim = sb.st_mtimespec;
+#elif defined HAVE_STRUCT_STAT_ST_MTIM
 		ep->mtim = sb.st_mtim;
+#else
+		ep->mtim.tv_sec = sb.st_mtime;
```
Comment (lichray) [inline common/exf.c:207]: Do we care?

File: CMakeLists.txt (hunk 1 above)
Comment (matijaskala) [reply CMakeLists.txt:146]: Not for me but see the commit message

File: common/exf.c (hunk 2 above)
Comment (matijaskala) [reply common/exf.c:207]: I could have simply put `#error` here but I thought it wasn't too much effort to implement this case.

**Pull request merged**